test/python/test_wrapper.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name
+
+"""Tests for the wrapper functionality."""
+
+import unittest
+
+import qiskit.wrapper
+from qiskit import QISKitError
+from .common import QiskitTestCase, requires_qe_access
+from .test_backends import remove_backends_from_list
+
+
+class TestWrapper(QiskitTestCase):
+    """Wrapper test case."""
+    @requires_qe_access
+    def test_wrapper_register_ok(self, QE_TOKEN, QE_URL, hub, group, project):
+        """Test wrapper.register()."""
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                provider_name='ibmq')
+        backends = qiskit.wrapper.available_backends()
+        backends = remove_backends_from_list(backends)
+        self.log.info(backends)
+        self.assertTrue(len(backends) > 0)
+
+    @requires_qe_access
+    def test_backends_with_filter(self, QE_TOKEN, QE_URL, hub, group, project):
+        """Test wrapper.available_backends(filter=...)."""
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                provider_name='ibmq')
+        backends = qiskit.wrapper.available_backends({'local': False,
+                                                      'simulator': True})
+        self.log.info(backends)
+        self.assertTrue(len(backends) > 0)
+
+    def test_local_backends(self):
+        """Test wrapper.local_backends(filter=...)."""
+        local_backends = qiskit.wrapper.local_backends()
+        self.log.info(local_backends)
+        self.assertTrue(len(local_backends) > 0)
+
+    @requires_qe_access
+    def test_register_twice(self, QE_TOKEN, QE_URL, hub, group, project):
+        """Test double registration of the same credentials."""
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project)
+        with self.assertRaises(QISKitError):
+            qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project)
+
+    @requires_qe_access
+    def test_register_twice_different_names(self, QE_TOKEN, QE_URL,
+                                            hub, group, project):
+        """Test double registration of same credentials but different names."""
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                provider_name='provider1')
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                provider_name='provider2')
+        self.assertEqual(['local', 'provider1', 'provider2'],
```

Comment:
Non-blocking: I'd relax the test and simply test if the names are in `registered_providers()`.

Reply:
Good catch - the tests needed a revamp! I'll make them to operate on the "relative" difference of providers registered at every point in time, in order to avoid having to take into account `local`.